File: client/src/redux/validation/event.js
function validateEvent(event) {
  const { title, location, starttime, endtime } = event;

  const errors = {};

  if (title.length === 0) errors.title = 'Title cannot be empty';
  if (location.length === 0) errors.location = 'Location cannot be empty';
  if (starttime.length === 0) errors.starttime = 'Start time cannot be empty';
  if (endtime.length === 0) errors.endtime = 'End time cannot be empty';
  if (Date.parse(starttime) > Date.parse(endtime)) errors.endtime = 'End cannot occur before start';

  return { errors, validated: Object.values(errors).length === 0 };
}

function validateUpdateEvent(event) {
  const { title, location, starttime, endtime } = event;

  const errors = {};

  if (!title && !location && !starttime && !endtime) errors.title = 'You have to update atleast one property';
  if (Date.parse(starttime) > Date.parse(endtime)) errors.endtime = 'End cannot occur before start';

  return { errors, validated: Object.values(errors).length === 0 };
}


export { validateEvent, validateUpdateEvent };

File: backend/validation/event.py
import datetime

def validate_put_event(title, location, starttime, endtime):
    errors = {}

    present = datetime.datetime.now()

    if(starttime is None): 
      errors['starttime'] = 'Start time is empty'
      return errors

    if(endtime is None):
       errors['endtime'] = 'End time is empty'
       return errors

    starttimeiso = datetime.datetime.strptime(starttime, "%Y-%m-%dT%H:%M:%S.%fZ")
    endtimeiso = datetime.datetime.strptime(endtime, "%Y-%m-%dT%H:%M:%S.%fZ")

    if(starttimeiso < present): errors['starttime'] = 'Earliest start time cannot be in the past'
    if(endtimeiso < present): errors['starttime'] = 'Earliest end time cannot be in the past'
    if(endtime < starttime): errors['endtime'] = 'Earliest end time cannot occur before earliest start time'

    if(title is None): errors['title'] = 'Title is empty'
    if(location is None): errors['location'] = 'Location is empty'

    return errors

def validate_post_event(title, location, starttime, endtime):
    errors = {}

    present = datetime.datetime.now()
    starttimeiso = datetime.datetime.strptime(starttime, "%Y-%m-%dT%H:%M:%S.%fZ")
    endtimeiso = datetime.datetime.strptime(endtime, "%Y-%m-%dT%H:%M:%S.%fZ")

    if(starttimeiso < present): errors['starttime'] = 'Earliest start time cannot be in the past'
    if(endtimeiso < present): errors['starttime'] = 'Earliest end time cannot be in the past'
    if(endtime < starttime): errors['endtime'] = 'Earliest end time cannot occur before earliest start time'

    if(title is None): errors['title'] = 'Title is empty'
    if(location is None): errors['location'] = 'Location is empty'
    if(starttime is None): errors['starttime'] = 'Start time is empty'
    if(endtime is None): errors['endtime'] = 'End time is empty'

    return errors